Assets/Newtonsoft.Json.UnityConverters.Tests/UnityContractResolverTests.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using UnityEngine;
+
+namespace Newtonsoft.Json.UnityConverters.Tests
+{
+    public class UnityContractResolverTests : TypeTester<UnityContractResolverTests.MockScriptableObject>
+    {
+        public static readonly IReadOnlyCollection<(MockScriptableObject deserialized, object anonymous)> representations = new (MockScriptableObject, object)[] {
+            (ScriptableObject.CreateInstance<MockScriptableObject>(), new ExpectedSignature {
+                float1 = 0f,
+                name = string.Empty,
+                hideFlags = HideFlags.None,
+                float2 = 0f,
+            }),
+            (CreateMockInstance("myObject", HideFlags.DontSaveInBuild | HideFlags.HideInHierarchy, 1, 2), new ExpectedSignature {
+                float1 = 1,
+                name = "myObject",
+                hideFlags = HideFlags.DontSaveInBuild | HideFlags.HideInHierarchy,
+                float2 = 2,
+            }),
+        };
+
+        protected override bool AreEqual([AllowNull] MockScriptableObject a, [AllowNull] MockScriptableObject b)
+        {
+            if (a is null && b is null)
+            {
+                return true;
+            }
+
+            if (a is null || b is null)
+            {
+                return false;
+            }
+
+            return a.name == b.name
+                && a.hideFlags == b.hideFlags
+                && a.float1 == b.float1
+                && a.GetFloat2() == b.GetFloat2();
+        }
+
+        private static MockScriptableObject CreateMockInstance(string name, HideFlags hideFlags, float float1, float float2)
+        {
+            MockScriptableObject instance = ScriptableObject.CreateInstance<MockScriptableObject>();
+
+            instance.name = name;
+            instance.hideFlags = hideFlags;
+            instance.float1 = float1;
+            instance.SetFloat2(float2);
+
+            return instance;
+        }
+
+        private struct ExpectedSignature
```

Comment:
Codacy found an issue: [Implement 'IEquatable<T>' in value type 'ExpectedSignature'.](https://app.codacy.com/manual/jilleJr/Newtonsoft.Json-for-Unity.Converters/pullRequest?prid=5855822)